Indicators/RelativeVolume.cs

```diff
@@ -0,0 +1,154 @@
+/*
+ * QUANTCONNECT.COM - Democratizing Finance, Empowering Individuals.
+ * Lean Algorithmic Trading Engine v2.0. Copyright 2014 QuantConnect Corporation.
+ * 
+ * Licensed under the Apache License, Version 2.0 (the "License"); 
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ * 
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+*/
+using System;
+using System.Linq;
+using System.Collections.Generic;
+using QuantConnect.Data.Market;
+
+namespace QuantConnect.Indicators
+{
+    /// <summary>
+    /// The Relative Volume indicator is an indicator that compares current 
+    /// cumulative volume to the cumulative volume for a given 
+    /// time of day, measured as a ratio.
+    /// 
+    /// Current volume from open to current time of day / Average over the past x days from open to current time of day
+    /// </summary>
+    public class RelativeVolume : TradeBarIndicator, IIndicatorWarmUpPeriodProvider
+    {
+        private readonly List<decimal> CurrentVolume = new List<decimal>();
+        private readonly Dictionary<DateTime, decimal> HistoricalVolumes = new Dictionary<DateTime, decimal>();
+        private DateTime _previousDay;
+
+        /// <summary>
+        /// Gets a flag indicating when the indicator is ready and fully initialized
+        /// </summary>
+        public override bool IsReady => Samples >= WarmUpPeriod;
+
+        /// <summary>
+        /// Required period, in data points, for the indicator to be ready and fully initialized.
+        /// </summary>
+        public int WarmUpPeriod { get; }
+
+        /// <summary>
+        /// Initializeds a new instance of the RelativeVolume class using the specified period
+        /// </summary>
+        /// <param name="period">The period over which to perform to computation</param>
+        public RelativeVolume(int period = 14)
+            : this($"RVOL({period})", period)
```

Comment:
What if the user creates the indicator with Minute resolution and a period of 10, I don't think it would behave as expected?

I think more generically the indicator could rely on the user provided period and **resolution** to determine to window time span to average volume on, right?

Reply:
Hi Martin, thank you for the feedback. I will start making these changes. If a user were to use a minute resolution and period of 10, an ArgumentException would be thrown stating that at least one day of data Is required. If the user were to use a minute resolution they would need a period of at least 390 minutes or one trading day. 

In terms of determining a window time span to average volume, it is based on days. Here is a drawing that @ikamanu drew clearing up a similar question I had: 

![image](https://user-images.githubusercontent.com/67654222/125892080-26a602ec-13dc-48f8-9d30-c6e0893400ac.png)

Reply:
Right 👍 my point was why not remove the hard coded `based on days` notion and just use the resolution the user provided as input passing it into the indicator, that could be the timespan?

>  If a user were to use a minute resolution and period of 10, an ArgumentException would be thrown stating that at least one day of data Is required

This wouldn't be very user friendly, cause also it's happening at runtime, after create the indicator while updating him?